packages/dev/loaders/src/OBJ/solidParser.ts

```diff
@@ -543,7 +550,35 @@ export class SolidParser {
         }
 
         // Split the file into lines
-        const lines = data.split("\n");
+        // Preprocess line data
+        const linesOBJ = data.split("\n");
+        const lines: string[] = [];
+        const lineLines: string[] = [];
+
+        for (let i = 0; i < linesOBJ.length; i++) {
+            const line = linesOBJ[i].trim().replace(/\s\s/g, " ");
+
+            // Comment or newLine
+            if (line.length === 0 || line.charAt(0) === "#") {
+                continue;
+            }
+
+            if (SolidParser._IsLineElement(line)) {
+                const lineValues = line.split(" ");
+                // create line elements with two vertices only
+                for (let i = 1; i < lineValues.length - 1; i++) {
+                    lineLines.push(`l ${lineValues[i]} ${lineValues[i + 1]}`);
+                }
+            } else {
+                lines.push(line);
+            }
+        }
+
+        for (let i = 0; i < lineLines.length; i++) {

```

Comment:
pushing at the end won t work with groups, can you replace the line instead ?

Reply:
> pushing at the end won t work with group

I've added support for the `g` and `o` element. Now it takes grouping into account.

Tested with:

```    
    v 0.000000 2.000000 2.000000
    v 0.000000 0.000000 2.000000
    v 2.000000 0.000000 2.000000
    v 2.000000 2.000000 2.000000
    v 0.000000 2.000000 0.000000
    v 0.000000 0.000000 0.000000
    v 2.000000 0.000000 0.000000
    v 2.000000 2.000000 0.000000
    # 8 vertices

    g top cube
    f 5 1 4 8
    f 1 2 3 4

    g front cube
    o kokki1
    l 1 2 3 4
    g back cube
    o kokki2
    l 8 7 6 5
    g right cube
    l 4 3 7 8
   
    g left cube
    l 5 6 2 1
    g bottom cube
    l 2 6 7 3
    # 6 elements
```